log4j-core/src/main/java/org/apache/logging/log4j/core/net/JndiManager.java

```diff
@@ -168,21 +206,89 @@ protected boolean releaseSub(final long timeout, final TimeUnit timeUnit) {
      * @throws  NamingException if a naming exception is encountered
      */
     @SuppressWarnings("unchecked")
-    public <T> T lookup(final String name) throws NamingException {
+    public synchronized <T> T lookup(final String name) throws NamingException {
+        try {
+            URI uri = new URI(name);
+            if (!allowedProtocols.contains(uri.getScheme().toLowerCase(Locale.ROOT))) {
+                LOGGER.warn("Log4j JNDI does not allow protocol {}", uri.getScheme());
+                return null;
+            }
+            if (LDAP.equalsIgnoreCase(uri.getScheme()) || LDAPS.equalsIgnoreCase(uri.getScheme())) {
+                if (!allowedHosts.contains(uri.getHost())) {
+                    LOGGER.warn("Attempt to access ldap server not in allowed list");
+                    return null;
+                }
+                Attributes attributes = this.context.getAttributes(name);
```

Comment:
Are we confident the attributes resolved here are equivalent to the results of `context.lookup`? I imagine the attributes could change over time.

Reply:
I assume you mean the attributes associated with the returned item. All the attributes returned from LDAP for Java objects are specified in https://datatracker.ietf.org/doc/html/rfc2713.

Reply:
I want to make sure that `context.getAttributes(name)` verifies against the same data (cached without any sort of refresh or network call) as the subsequent call to `context.lookup(name)`, otherwise the verification isn't very helpful.

Reply:
Well, yeah. That is how DirContext is documented to work. https://docs.oracle.com/javase/8/docs/api/javax/naming/directory/DirContext.html#getAttributes-java.lang.String-

Reply:
Perfect, thank you for the explanation, I'm not very familiar with these components :-)